internal/engine/image_build_and_deployer_test.go

```diff
@@ -536,6 +537,61 @@ func TestDeployInjectsOverrideCommand(t *testing.T) {
 	assert.Empty(t, c.Args)
 }
 
+func (f *ibdFixture) firstPodTemplateSpecHash() k8s.PodTemplateSpecHash {
+	entities, err := k8s.ParseYAMLFromString(f.k8s.Yaml)
+	if err != nil {
+		f.T().Fatal(err)
+	}
+
+	require.Equal(f.T(), 1, len(entities), "expected only one entity. Yaml contained: %s", f.k8s.Yaml)
+
+	d := entities[0].Obj.(*v1.Deployment)
```

Comment:
maybe `d, ok :=` here and `require.True(t, ok, "expected entity to be a deployment")` -- if it's NOT a deployment for some reason, this is easier to debug than the weird nil pointer errors you'll get otherwise

Reply:
(but also why are you assuming the first entity is always a deployment?)